syscalls.go

```diff
@@ -18,13 +18,22 @@ import (
 var ErrNotExist = os.ErrNotExist
 
 // RemoveMemlockRlimit removes the limit on the amount of memory the current
-// process can lock into RAM. Returns a function that restores the limit to
-// its previous value.
+// process can lock into RAM, if necessary.
 //
 // This is not required to load eBPF resources on kernel versions 5.11+
-// due to the introduction of cgroup-based memory accounting.
-func RemoveMemlockRlimit() (func() error, error) {
-	return unix.RemoveMemlockRlimit()
+// due to the introduction of cgroup-based memory accounting. On such kernels
+// the function is a no-op.
+//
+// Since the function changes global per-process limits it should be invoked at
+// program start up, in main() or init().
+//
+// This function exists as a convenience and doesn't cover situations where
```

Comment:
Use of a double negative here makes it hard to understand.

`... as convenience and should only be used when permanently raising RLIMIT_MEMLOCK to infinite is appropriate.`

When is it not appropriate? Should `prlimit(2)` be used in case it isn't?